src/complexity_measures/sample_entropy.jl

```diff
@@ -0,0 +1,144 @@
+using DelayEmbeddings
+using Statistics
+using Neighborhood.NearestNeighbors: inrangecount, Chebyshev
+
+export SampleEntropy
+
+"""
+    SampleEntropy(; r::Real = 0.2 * Statistics.std(x), m::Int = 2, τ::Int = 1,
+        metric = Chebyshev()
+    SampleEntropy(x::AbstractVector; m = 2, τ = 1, metric = Chebyshev())
+
+An estimator for the sample entropy complexity measure (Richman & Moorman,
+2000)[^Richman2000], used with [`complexity`](@ref) and [`complexity_normalized`](@ref).
+
+Providing a univariate timeseries `x` to the constructors automatically determines the
+radius `r` as `0.2 * Statistics.std(x)`.
+
+## Description
+
+Sample entropy is defined as
+
+```math
+SampEn(m, r) = \\lim_{N \\to \\infty} \\left[ -\\ln \\dfrac{A^{m+1}(r)}{B^m(r)} \\right],
+```
+
+An *estimator* for sample entropy using radius `r`, embedding dimension `m`,
+embedding lag `τ` is
+
+```math
+SampEn(m,r, N) = -\\ln{\\dfrac{A(r, N)}{B(r, N)}}.
+```
+
+Here,
+
+```math
+\\begin{aligned}
+B(r, m, N) = \\sum_{i = 1}^{N-m\\tau} \\sum_{j = 1, j \\neq i}^{N-m\\tau} \\theta(d({\\bf x}_i^m, {\\bf x}_j^m) \\leq r) \\\\
+A(r, m, N) = \\sum_{i = 1}^{N-m\\tau} \\sum_{j = 1, j \\neq i}^{N-m\\tau} \\theta(d({\\bf x}_i^{m+1}, {\\bf x}_j^{m+1}) \\leq r) \\\\
+\\end{aligned},
+```
+
+where ``\\theta(\\cdot)`` returns 1 if the argument is true and 0 otherwise,
+and ``d(x, y)`` computes the distance between ``x`` and ``y`` according to `metric`
+(default is Chebyshev), and  ``{\\bf x}_i^{m}`` and ``{\\bf x}_i^{m+1}`` are `m`-dimensional and
+`m+1`-dimensional embedding vectors, where `k`-dimensional embedding vectors are constructed
+from the input timeseries ``x(t)`` as
+
+```math
+{\\bf x}_i^k = (x(i), x(i+τ), x(i+2τ), \\ldots, x(i+(k-1)\\tau)).
+```
+
+Quoting Richman & Moorman (2002): "SampEn(m,r,N) will be defined except when B = 0,
+in which case no regularity has been detected, or when A = 0, which corresponds to
+a conditional probability of 0 and an infinite value of SampEn(m,r,N)".
+In these cases, `NaN` is returned.
+
+If computing the normalized measure, then the resulting sample entropy is on `[0, 1]`.
+
+!!! note "Flexible embedding lag"
+    The original algorithm fixes `τ = 1`. All formulas here are modified to account for
+    any `τ`.
+
+[^Richman2000]: Richman, J. S., & Moorman, J. R. (2000). Physiological time-series
+    analysis using approximate entropy and sample entropy. American Journal of
+    Physiology-Heart and Circulatory Physiology, 278(6), H2039-H2049.
+"""
+Base.@kwdef struct SampleEntropy{I, R, M} <: ComplexityMeasure
+    m::I = 2
+    τ::I = 1
+    metric::M = Chebyshev()
+    r::R = 0.1
```

Comment:
see comment above. I don't think this `r` is a justified choice without giving x